src/libvirtApi/domain.js

```diff
@@ -328,7 +329,11 @@ export function domainCreate({
                         args.sourceType = LOCAL_INSTALL_MEDIA_SOURCE;
                         args.source = downloadDir + filename;
 
-                        return downloadRhelImage(accessToken, url, filename, downloadDir, isSystem);
+                        return downloadRhelImage(accessToken, url, filename, downloadDir, isSystem)
+                                .stream(progress => {
+                                    const percentages = progress.trim().split('\n');
+                                    updateImageDownloadProgress(vmName, connectionName, percentages[percentages.length - 1]);
```

Comment:
Of course complete precision does not matter here, but reading from a stream is a generic skill, so for the sake of teaching: It can happen that some packet gets transfered with a boundary that is in the middle of a number. The clean way of reading a line-separated stream is to *not* do the trim(), but:

 - add the currently read block to a string buffer.
 - split the buffer on `\n`; keep the last part in the buffer
 - process the first n-1 parts (i.e. the ones that got terminated); of course for your case you only need the most recent complete percentage, i.e. parts[n-2]

Have a look at [c-podman's monitor](https://github.com/cockpit-project/cockpit-podman/blob/18fb5a2f9e702e3447dceacec8bdb951b36351f4/src/rest.js#L25) how to do this correctly and elegantly.